hoss/coordinate_utilities.py

```diff
@@ -0,0 +1,353 @@
+""" This module contains utility functions used for
+    coordinate variables and methods to convert the
+    coordinate variable data to x/y dimension scales
+"""
+
+from typing import Dict, Set, Tuple
+
+import numpy as np
+from netCDF4 import Dataset
+from numpy import ndarray
+from pyproj import CRS
+from varinfo import VariableFromDmr, VarInfoFromDmr
+
+from hoss.exceptions import (
+    IrregularCoordinateDatasets,
+    MissingCoordinateDataset,
+    MissingValidCoordinateDataset,
+)
+from hoss.projection_utilities import (
+    get_x_y_extents_from_geographic_points,
+)
+
+
+def get_override_projected_dimension_name(
+    varinfo: VarInfoFromDmr,
+    variable_name: str,
+) -> str:
+    """returns the x-y projection variable names that would
+    match the geo coordinate names. The `latitude` coordinate
+    variable name gets converted to 'projected_y' dimension scale
+    and the `longitude` coordinate variable name gets converted to
+    'projected_x'
+
+    """
+    override_variable = varinfo.get_variable(variable_name)
+    projected_dimension_name = ''
```

Comment:
I think I had to declare it above as default value to keep the variable in scope for the return call

Reply:
I don't think you've understood my comment - you have two choices here: declare it first, or declare it in an `else`. I am asking you to be consistent to the rest of the style of the repository and put the default value in an `else` statement.

To avoid two `else` statements, you could restructure the logic below as:

```
if override_variable is not None and override_variables.is_latitude():
    projected_dimension_name = 'projected_y'
elif override_variable is not None and override_variable.is_longitude():
    projected_dimension_name = 'projected_x'
else:
    projected_dimension_name = ''
```

I still have two other concerns here:

1) Returning an empty string seems weird.
2) The projected dimension name should be distinct in the case of two grids. Using just `projected_x` and `projected_y` will cause clashes for those SMAP L3 products with both global and polar grids. (This _must_ be resolved before merging this PR)

Reply:
- corrected the names to add the group name (in next commit)
- simplified the structure to not individually check for lat/lon (in next commit)
- about returning an empty string..-technically  it should never be called from our code unless we have the coordinates. I could throw an exception - just seemed overkill